From ebf5eb1446e2d2b94610a3fccc12eddaae239e55 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:40 -0400
Subject: [tools] perf/tests: Use lower sample_freq in sw clock event period test

Message-id: <1408462094-14194-292-git-send-email-jolsa@redhat.com>
Patchwork-id: 88290
O-Subject: [PATCH RHEL7.1 BZ1131394 291/325] perf tests: Use lower sample_freq in sw clock event period test
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 67c1e4a53b17894e6a24f95057cc374c4be051cb
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Mon Nov 11 16:33:18 2013 -0300

description
===========
We were using it at 10 kHz, which doesn't work in machines where somehow
the max freq was auto reduced by the kernel:

[root@ssdandy ~]# perf test 19
19: Test software clock events have valid period values    : FAILED!
[root@ssdandy ~]# perf test -v 19
19: Test software clock events have valid period values    :
--- start ---
Couldn't open evlist: Invalid argument
---- end ----
Test software clock events have valid period values: FAILED!
[root@ssdandy ~]#

[root@ssdandy ~]# cat /proc/sys/kernel/perf_event_max_sample_rate
7000

Reducing it to 500 Hz should be good enough for this test and also
shouldn't affect what it is testing.

But warn the user if it fails, informing the knob and the freq tried.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/sw-clock.c b/tools/perf/tests/sw-clock.c
index ed77772..93a7139 100644
--- a/tools/perf/tests/sw-clock.c
+++ b/tools/perf/tests/sw-clock.c
@@ -34,7 +34,7 @@ static int __test__sw_clock_freq(enum perf_sw_ids clock_id)
   .freq = 1,
  };
 
- attr.sample_freq = 10000;
+ attr.sample_freq = 500;
 
  evlist = perf_evlist__new();
  if (evlist == NULL) {
@@ -58,8 +58,11 @@ static int __test__sw_clock_freq(enum perf_sw_ids clock_id)
  }
 
  if (perf_evlist__open(evlist)) {
+  const char *knob = "/proc/sys/kernel/perf_event_max_sample_rate";
+
   err = -errno;
-  pr_debug("Couldn't open evlist: %s\n", strerror(errno));
+  pr_debug("Couldn't open evlist: %s\nHint: check %s, using %" PRIu64 " in this test.\n",
+    strerror(errno), knob, (u64)attr.sample_freq);
   goto out_delete_maps;
  }
 
-- 
1.7.1